betamax-tests/src/test/groovy/software/betamax/tape/ContentEncodingSpec.groovy

```diff
@@ -47,7 +49,13 @@ class ContentEncodingSpec extends Specification {
         def response = new BasicResponse(HTTP_OK, "OK")
         response.addHeader(CONTENT_TYPE, PLAIN_TEXT_UTF_8.withoutParameters().toString())
         response.addHeader(CONTENT_ENCODING, encoding)
-        response.body = "O HAI!".bytes
+        if (encoding == "gzip") {
+            response.body = new GzipEncoder().encode("O HAI!")
+        } else if (encoding == "deflate") {
+            response.body = new DeflateEncoder().encode("O HAI!")
+        } else {
+            response.body = "O HAI!".bytes
+        }
```

Comment:
@faucct I agree a switch looks better. I was following prior art in the code until I have the time to do a full refactor. This was simply something to drop into place to get a SNAPSHOT out for you to use.